hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -253,14 +254,17 @@ protected void handleDeleteDirEvent(OmDirectoryInfo directoryInfo,
     }
   }
 
-  protected boolean flushAndCommitNSToDB(Map<Long, NSSummary> nsSummaryMap) {
-    try {
-      updateNSSummariesToDB(nsSummaryMap, Collections.emptyList());
-    } catch (IOException e) {
-      LOG.error("Unable to write Namespace Summary data in Recon DB.", e);
-      return false;
-    } finally {
-      nsSummaryMap.clear();
+  protected synchronized boolean flushAndCommitNSToDB(Map<Long, NSSummary> nsSummaryMap,
+                                                      long nsSummaryFlushToDBMaxThreshold) {
+    if (nsSummaryMap.size() >= nsSummaryFlushToDBMaxThreshold) {
```

Comment:
we should not have this check at this point, caller is already having this check